src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,142 @@
+        this.loadFactor = loadFactor;
+        this.threshold = (int) (initialCapacity * loadFactor);
+        this.size = 0;
+    }
+
     public void put(K key, V value) {
+        int hash = hash(key);
+        int index = indexFor(hash, table.length);
 
+        for (Node<K, V> node = table[index]; node != null; node = node.next) {
+            if (node.hash == hash && (Objects.equals(key, node.key))) {
+                node.value = value;
+                return;
+            }
+        }
+        addNode(hash, key, value, index);
     }
 
-    @Override
     public V getValue(K key) {
+        int hash = hash(key);
+        int index = indexFor(hash, table.length);
+
+        for (Node<K, V> node = table[index]; node != null; node = node.next) {
+            if (node.hash == hash && (Objects.equals(key, node.key))) {
+                return node.value;
+            }
+        }
+
         return null;
     }
 
-    @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void addNode(int hash, K key, V value, int index) {
+        Node<K, V> node = table[index];
+        table[index] = new Node<>(hash, key, value, node);
+        size++;
+
+        if (size >= threshold) {
+            resize();
+        }
+    }
+
+    private void resize() {
+        Node<K, V>[] oldTable = table;
+        int oldCapacity = oldTable.length;
+
+        int newCapacity = oldCapacity * 2;
+        Node<K, V>[] newTable = new Node[newCapacity];
+        threshold = (int) (newCapacity * loadFactor);
+
+        for (int i = 0; i < oldCapacity; i++) {
+            for (Node<K, V> node = oldTable[i]; node != null; ) {
+                Node<K, V> next = node.next;
+
+                int index = indexFor(node.hash, newCapacity);
+                node.next = newTable[index];
+                newTable[index] = node;
+
+                node = next;
+            }
+        }
+
+        table = newTable;
+    }
+
+    private int hash(K key) {
+        return key == null ? 0 : key.hashCode();
+    }
+
+    private int indexFor(int hash, int length) {
+        return hash & (length - 1);
+    }
+
+    private static class Node<K, V> {
+        private final int hash;
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+
+        public int getHash() {
+            return hash;
+        }
+
+        public K getKey() {
+            return key;
+        }
+
+        public V getValue() {
+            return value;
+        }
+
+        public void setValue(V value) {
+            this.value = value;
+        }
+
+        public Node<K, V> getNext() {
+            return next;
+        }
+
+        public void setNext(Node<K, V> next) {
+            this.next = next;
+        }
```

Comment:
Don’t use getters and setters in class Node
We have access to private fields of inner class, so using getters or setters is redundant.